# **0. Importação das Bibliotecas**

In [1]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

#plt.rc('figure', figsize=(5, 5))

# **1.0 Análise exploratória**

### **1.1 Carregamento dos Arquivos:**

**OBS:** 
A partir do ano de 2017 até 2020 teve acréscimo de geolocalização das tabelas

- **Base de dados por ocorrência:**

In [2]:
cols = ['id','data_inversa','dia_semana','horario','uf','br','causa_acidente','tipo_acidente','fase_dia',
          'condicao_metereologica','tracado_via','uso_solo','pessoas','mortos','feridos_leves',
          'feridos_graves','ilesos','feridos','veiculos']


df_ocorrencia_2007 = pd.read_csv('archive/por_ocorrencias/datatran2007.csv', sep=';', engine='python', encoding='latin-1', usecols=cols)
df_ocorrencia_2008 = pd.read_csv('archive/por_ocorrencias/datatran2008.csv', sep=';', engine='python', encoding='latin-1', usecols=cols)
df_ocorrencia_2009 = pd.read_csv('archive/por_ocorrencias/datatran2009.csv', sep=';', engine='python', encoding='latin-1', usecols=cols)
df_ocorrencia_2010 = pd.read_csv('archive/por_ocorrencias/datatran2010.csv', sep=';', engine='python', encoding='latin-1', usecols=cols)
df_ocorrencia_2011 = pd.read_csv('archive/por_ocorrencias/datatran2011.csv', sep=';', engine='python', encoding='latin-1', usecols=cols)
df_ocorrencia_2012 = pd.read_csv('archive/por_ocorrencias/datatran2012.csv', sep=';', engine='python', encoding='latin-1', usecols=cols)
df_ocorrencia_2013 = pd.read_csv('archive/por_ocorrencias/datatran2013.csv', sep=';', engine='python', encoding='latin-1', usecols=cols)
df_ocorrencia_2014 = pd.read_csv('archive/por_ocorrencias/datatran2014.csv', sep=';', engine='python', encoding='latin-1', usecols=cols)
df_ocorrencia_2015 = pd.read_csv('archive/por_ocorrencias/datatran2015.csv', sep=';', engine='python', encoding='latin-1', usecols=cols)
df_ocorrencia_2016 = pd.read_csv('archive/por_ocorrencias/datatran2016.csv', sep=';', engine='python', encoding='latin-1', usecols=cols)
df_ocorrencia_2017 = pd.read_csv('archive/por_ocorrencias/datatran2017.csv', sep=';', engine='python', encoding='latin-1', usecols=cols + ['latitude','longitude'])
df_ocorrencia_2018 = pd.read_csv('archive/por_ocorrencias/datatran2018.csv', sep=';', engine='python', encoding='latin-1', usecols=cols + ['latitude','longitude'])
df_ocorrencia_2019 = pd.read_csv('archive/por_ocorrencias/datatran2019.csv', sep=';', engine='python', encoding='latin-1', usecols=cols + ['latitude','longitude'])
df_ocorrencia_2020 = pd.read_csv('archive/por_ocorrencias/datatran2020.csv', sep=';', engine='python', encoding='latin-1', usecols=cols + ['latitude','longitude'])

- **Cabeçalho**

Visualizar os 10 primeiros registros do conjunto de dados

In [3]:
df_ocorrencia_2007.head(10).T

0                 1               2  \
id                                    10                10         1032898   
data_inversa                  11/06/2007        11/06/2007      13/08/2007   
dia_semana                       Segunda           Segunda         Segunda   
horario                         15:30:00          15:30:00        14:25:00   
uf                                    MG                MG              MG   
br                                   381               381              40   
causa_acidente          Falta de atenção  Falta de atenção          Outras   
tipo_acidente            Colisão frontal   Colisão frontal  Saída de Pista   
fase_dia                       Pleno dia         Pleno dia       Pleno dia   
condicao_metereologica         Ceu Claro             Chuva       Ceu Claro   
tracado_via                         Reta              Reta            Reta   
uso_solo                           Rural             Rural           Rural   
pessoas                                9                 9               3   
mortos                                 0                 0               0   
feridos_leves                          4                 4               0   
feridos_graves                         0                 0               1   
ilesos                                 5                 5               2   
feridos                                4                 4               1   
veiculos                               6                 6               1   

                                              3                            4  \
id                                      1051130                      1066824   
data_inversa                         12/02/2007                   20/11/2007   
dia_semana                              Segunda                        Terça   
horario                                02:10:00                     05:30:00   
uf                                           MA                           CE   
br                                          135                          222   
causa_acidente                 Animais na Pista  Defeito mecânico em veículo   
tipo_acidente           Atropelamento de animal                  Capotamento   
fase_dia                            Plena noite                    Amanhecer   
condicao_metereologica                Ceu Claro                    Ceu Claro   
tracado_via                                Reta                         Reta   
uso_solo                                 Urbano                        Rural   
pessoas                                       5                            1   
mortos                                        2                            0   
feridos_leves                                 2                            1   
feridos_graves                                1                            0   
ilesos                                        0                            0   
feridos                                       3                            1   
veiculos                                      1                            1   

                                  5                6                        7  \
id                          1069918          1070971                  1137914   
data_inversa             16/12/2007       05/03/2007               12/08/2007   
dia_semana                  Domingo          Segunda                  Domingo   
horario                    17:40:00         08:10:00                 22:30:00   
uf                               MA               PR                       ES   
br                              230              277                      101   
causa_acidente               Outras           Outras                   Outras   
tipo_acidente           Capotamento  Colisão lateral  Atropelamento de pessoa   
fase_dia                  Anoitecer        Pleno dia              Plena noite   
condicao_metereologica        Chuva        Ceu Claro               

In [4]:
df_ocorrencia_2017.head(10).T

0  \
id                                                 8   
data_inversa                              2017-01-01   
dia_semana                                   domingo   
horario                                     00:00:00   
uf                                                PR   
br                                               376   
causa_acidente                 Fenômenos da Natureza   
tipo_acidente           Queda de ocupante de veículo   
fase_dia                                 Plena Noite   
condicao_metereologica                         Chuva   
tracado_via                                     Reta   
uso_solo                                         Não   
pessoas                                            1   
mortos                                             0   
feridos_leves                                      0   
feridos_graves                                     1   
ilesos                                             0   
feridos                                            1   
veiculos                                           1   
latitude                                -23,09880731   
longitude                               -52,38789369   

                                                  1                 2  \
id                                                9                11   
data_inversa                             2017-01-01        2017-01-01   
dia_semana                                  domingo           domingo   
horario                                    00:01:00          00:00:00   
uf                                               SC                PR   
br                                              101               153   
causa_acidente          Falta de Atenção à Condução  Animais na Pista   
tipo_acidente           Colisão com objeto estático       Capotamento   
fase_dia                                Plena Noite       Plena Noite   
condicao_metereologica                        Chuva    Garoa/Chuvisco   
tracado_via                                   Curva              Reta   
uso_solo                                        Não               Não   
pessoas                                           1                 2   
mortos                                            0                 0   
feridos_leves                                     0                 2   
feridos_graves                                    0                 0   
ilesos                                            1                 0   
feridos                                           0                 2   
veiculos                                          1                 1   
latitude                                   -27,8101      -23,36951985   
longitude                                  -48,6357      309,93513107   

                                                              3  \
id                                                           12   
data_inversa                                         2017-01-01   
dia_semana                                              domingo   
horario                                                00:00:00   
uf                                                           GO   
br                                                          153   
causa_acidente          Avarias e/ou desgaste excessivo no pneu   
tipo_acidente                                        Tombamento   
fase_dia                                            Plena Noite   
condicao_metereologica                                Céu Claro   
tracado_via                                                Reta   
uso_solo                                                    Sim   
pessoas                                                       1   
mortos                                                        0   
feridos_leves                                                 0   
feridos_graves                                                1   
ilesos                                                        0   
ferid

- **Adicionar coluna ```ano``` em todos os dataframes:**

In [5]:
df_ocorrencia_2007['ano'] = 2007
df_ocorrencia_2008['ano'] = 2008
df_ocorrencia_2009['ano'] = 2009
df_ocorrencia_2010['ano'] = 2010
df_ocorrencia_2011['ano'] = 2011
df_ocorrencia_2012['ano'] = 2012
df_ocorrencia_2013['ano'] = 2013
df_ocorrencia_2014['ano'] = 2014
df_ocorrencia_2015['ano'] = 2015
df_ocorrencia_2016['ano'] = 2016
df_ocorrencia_2017['ano'] = 2017
df_ocorrencia_2018['ano'] = 2018
df_ocorrencia_2019['ano'] = 2019
df_ocorrencia_2020['ano'] = 2020

- **Concatenar todos os DataFrames**

In [6]:
df = pd.concat([df_ocorrencia_2007,df_ocorrencia_2008,df_ocorrencia_2009,df_ocorrencia_2010,
                                 df_ocorrencia_2011,df_ocorrencia_2012,df_ocorrencia_2013,df_ocorrencia_2014,
                                 df_ocorrencia_2015,df_ocorrencia_2016,df_ocorrencia_2017,df_ocorrencia_2018,
                                 df_ocorrencia_2019,df_ocorrencia_2020], sort = False)

- **Filtragem pelo escopo do Rio Grande do Sul, uf(RS)**

In [7]:
df= df[df['uf'] == 'RS']

### **1.2 Descrição dos dados:**

Visualizar as colunas e os respectivos tipos de dados que elas estão classificadas

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 143098 entries, 13 to 45243
Data columns (total 22 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   id                      143098 non-null  float64
 1   data_inversa            143098 non-null  object 
 2   dia_semana              143098 non-null  object 
 3   horario                 143098 non-null  object 
 4   uf                      143098 non-null  object 
 5   br                      143045 non-null  object 
 6   causa_acidente          143098 non-null  object 
 7   tipo_acidente           143098 non-null  object 
 8   fase_dia                143098 non-null  object 
 9   condicao_metereologica  143098 non-null  object 
 10  tracado_via             143098 non-null  object 
 11  uso_solo                143098 non-null  object 
 12  pessoas                 143098 non-null  int64  
 13  mortos                  143098 non-null  int64  
 14  feridos_leves       

**Alguns pontos observaveis:**

- data inversa deveria ser `date`

- Latitude e Longitude deveriam ser `float` e estão como object (string)

- Horario deveria ser `time`

- **Quantidade de linhas e colunas**

In [9]:
print(f"{df.shape[0]} linhas e {df.shape[1]} colunas.")

143098 linhas e 22 colunas.


#### **1.2.1 breve limpeza dos dados**

- **Eliminar linhas duplicadas do dataframe:**

In [10]:
df.drop_duplicates(subset='id', keep= 'first', inplace=True)

In [11]:
print(f"Após a remoção dos registros duplicados: {df.shape[0]} linhas e {df.shape[1]} colunas.")

Após a remoção dos registros duplicados: 142496 linhas e 22 colunas.


- **_missing_ values**

In [12]:
# Porcentagem de valores faltantes
(df.isnull().sum()/df.shape[0]).sort_values()

id                        0.000000
ano                       0.000000
veiculos                  0.000000
feridos                   0.000000
ilesos                    0.000000
feridos_graves            0.000000
feridos_leves             0.000000
mortos                    0.000000
pessoas                   0.000000
uso_solo                  0.000000
tracado_via               0.000000
condicao_metereologica    0.000000
fase_dia                  0.000000
tipo_acidente             0.000000
causa_acidente            0.000000
uf                        0.000000
horario                   0.000000
dia_semana                0.000000
data_inversa              0.000000
br                        0.000358
latitude                  0.874768
longitude                 0.874768
dtype: float64

- #### **Eliminar linhas com todos os valores incompletos ou nulos:**

In [13]:
df.dropna(how='all', inplace=True)
print(f"Após a remoção dos registros nulos: {df.shape[0]} linhas e {df.shape[1]} colunas.")

Após a remoção dos registros nulos: 142496 linhas e 22 colunas.


- **Reindexar os registros, por conta da remoção das duplicatas**

In [14]:
df.reset_index(drop=True, inplace=True)

### **1.2.2 Visualização dos valores possíveis**

Vamos verificar os valores das variáveis categóricas:

- **`dia_semana`**
- **`uf`**
- **`br`**
- **`causa_acidente`**
- **`tipo_acidente`**
- **`fase_dia`**
- **`condicao_metereologica`**
- **`tracado_via`**

e algumas visualizações sobre tendências nas variáveis numéricas como:
- **`veiculos`**
- **`mortos`**

In [15]:
df['dia_semana'].unique()

array(['Segunda', 'Terça', 'Quarta', 'Quinta', 'Sexta', 'Sábado',
       'Domingo', 'domingo', 'segunda-feira', 'terça-feira',
       'quarta-feira', 'quinta-feira', 'sexta-feira', 'sábado'],
      dtype=object)

In [16]:
df['uf'].unique()

array(['RS'], dtype=object)

In [17]:
df['br'].unique()

array(['158', '386', '473', '290', '101', '116', '392', '287', '472',
       '293', '153', '471', '285', '468', '480', '377', '604', 116, 290,
       392, 285, 287, 386, 472, 377, 158, 293, 101, 471, 153, 468, 480,
       473, 470, '470', '0', '178', '453', '183', 453, 448, nan],
      dtype=object)

In [18]:
df['causa_acidente'].unique()

array(['Não guardar distância de segurança', 'Falta de atenção', 'Outras',
       'Ultrapassagem indevida', 'Defeito mecânico em veículo',
       'Ingestão de álcool', 'Animais na Pista', 'Dormindo',
       'Velocidade incompatível', 'Desobediência à sinalização',
       'Defeito na via', 'Defeito Mecânico no Veículo',
       'Ingestão de Álcool', 'Condutor Dormindo',
       'Falta de Atenção à Condução',
       'Desobediência às normas de trânsito pelo condutor',
       'Ultrapassagem Indevida', 'Falta de Atenção do Pedestre',
       'Sinalização da via insuficiente ou inadequada',
       'Avarias e/ou desgaste excessivo no pneu',
       'Velocidade Incompatível', 'Pista Escorregadia',
       'Fenômenos da Natureza', 'Mal Súbito',
       'Carga excessiva e/ou mal acondicionada', 'Defeito na Via',
       'Objeto estático sobre o leito carroçável',
       'Restrição de Visibilidade',
       'Deficiência ou não Acionamento do Sistema de Iluminação/Sinalização do Veículo',
       'Ingestã

In [19]:
df['tipo_acidente'].unique()

array(['Colisão traseira', 'Colisão Transversal', 'Colisão lateral',
       'Tombamento', 'Colisão frontal', 'Saída de Pista', 'Incêndio',
       'Colisão com objeto fixo',
       'Queda de motocicleta / bicicleta / veículo',
       'Colisão com bicicleta', 'Atropelamento de pessoa', 'Capotamento',
       'Colisão com objeto móvel', 'Atropelamento de animal',
       'Derramamento de Carga', 'Danos Eventuais',
       'Atropelamento de Animal', 'Saída de leito carroçável',
       'Colisão transversal', 'Atropelamento de Pedestre',
       'Colisão com objeto estático', 'Engavetamento', 'Danos eventuais',
       'Queda de ocupante de veículo', 'Colisão com objeto em movimento',
       'Derramamento de carga'], dtype=object)

In [20]:
df['fase_dia'].unique()

array(['Plena noite', 'Pleno dia', 'Anoitecer', 'Amanhecer', '(null)',
       'Plena Noite'], dtype=object)

In [21]:
df['condicao_metereologica'].unique()

array(['Ceu Claro', 'Sol', 'Vento', 'Chuva', 'Nublado',
       'Nevoeiro/neblina', 'Ignorada', '(null)', 'Neve', 'Granizo',
       'Céu Claro', 'Garoa/Chuvisco', 'Nevoeiro/Neblina', 'Ignorado'],
      dtype=object)

In [22]:
df['tracado_via'].unique()

array(['Reta', 'Curva', 'Cruzamento', '(null)', 'Não Informado',
       'Rotatória', 'Desvio Temporário', 'Interseção de vias', 'Ponte',
       'Retorno Regulamentado', 'Viaduto', 'Túnel'], dtype=object)

In [23]:
df['veiculos'].describe()

count    142496.000000
mean          1.718771
std           0.681115
min           1.000000
25%           1.000000
50%           2.000000
75%           2.000000
max          15.000000
Name: veiculos, dtype: float64

In [24]:
df['mortos'].describe()

count    142496.000000
mean          0.038085
std           0.238218
min           0.000000
25%           0.000000
50%           0.000000
75%           0.000000
max          13.000000
Name: mortos, dtype: float64

- **Análise dos valores**

---
**`dia_semana`** temos os valores: 
    
    'Segunda', 'Terça', 'Domingo', 'Quinta', 'Quarta', 'Sexta',
       'Sábado', 'domingo', 'segunda-feira', 'terça-feira', 'sexta-feira',
       'quarta-feira', 'quinta-feira', 'sábado'. Como tem alguns duplicados, os valores estão distintos, os reais são: 

* "Segunda-feira", 
* "Terça-feira", 
* "Quarta-feira" 
* "Quinta-feira" 
* "Sexta-feira" 
* "Sábado" 
* "Domingo". 
___
**`uf`**:
Na uf temos as siglas dos estados do Brasil, há apenas a string '(null)' que deve ser removida;
___
**`br`**:

'158', '386', '473', '290', '101', '116', '392', '287', '472',
       '293', '153', '471', '285', '468', '480', '377', '604', 116, 290,
       392, 285, 287, 386, 472, 377, 158, 293, 101, 471, 153, 468, 480,
       473, 470, '470', '0', '178', '453', '183', 453, 448, nan
       
Nela temos os números da br onde ocorreram os acidentes. Há o valor '0', que pode significar nomenclatura antiga.

As rodovias sequenciais compreendem a nomenclatura antiga, e portanto deve-se normalizá-las. 
Há uma [tabela para conversão dos valores](https://pt.wikipedia.org/wiki/Sistema_antigo_de_numera%C3%A7%C3%A3o_de_rodovias_federais_do_Brasil). Embora os valores até 105 tenham alguma tradução para a nomenclatura atual, a br '0' parece ser um _outlier_.
___
**`causa_acidente`**:

'Falta de atenção', 'Outras', 'Animais na Pista',
       'Defeito mecânico em veículo',
       'Não guardar distância de segurança', 'Velocidade incompatível',
       'Desobediência à sinalização', 'Ingestão de álcool',
       'Defeito na via', 'Dormindo', 'Ultrapassagem indevida', '(null)',
       'Desobediência às normas de trânsito pelo condutor',
       'Velocidade Incompatível', 'Ingestão de Álcool',
       'Restrição de Visibilidade', 'Falta de Atenção à Condução',
       'Defeito Mecânico no Veículo', 'Falta de Atenção do Pedestre',
       'Condutor Dormindo', 'Pista Escorregadia',
       'Ultrapassagem Indevida', 'Defeito na Via',
       'Avarias e/ou desgaste excessivo no pneu',
       'Sinalização da via insuficiente ou inadequada',
       'Fenômenos da Natureza', 'Mal Súbito',
       'Carga excessiva e/ou mal acondicionada',
       'Objeto estático sobre o leito carroçável',
       'Deficiência ou não Acionamento do Sistema de Iluminação/Sinalização do Veículo',
       'Ingestão de Substâncias Psicoativas', 'Agressão Externa',
       'Desobediência às normas de trânsito pelo pedestre',
       'Ingestão de álcool e/ou substâncias psicoativas pelo pedestre'.
       
Dentre esses valores, há alguns que divergem na escrita, porém são a mesma coisa, como 'Falta de atenção' e 'Falta de Atenção à Condução'. 'Defeito mecânico em veículo' e 'Defeito Mecânico no Veículo'.

___
**`tipo_acidente`**:
'Colisão frontal', 'Saída de Pista', 'Atropelamento de animal',
       'Capotamento', 'Colisão lateral', 'Atropelamento de pessoa',
       'Colisão traseira', 'Colisão Transversal', 'Tombamento',
       'Colisão com objeto fixo', 'Danos Eventuais',
       'Queda de motocicleta / bicicleta / veículo',
       'Derramamento de Carga', 'Colisão com bicicleta',
       'Colisão com objeto móvel', 'Incêndio', 'Atropelamento de Animal',
       'Saída de leito carroçável', 'Colisão com objeto estático',
       'Atropelamento de Pedestre', 'Queda de ocupante de veículo',
       'Colisão transversal', 'Colisão com objeto em movimento',
       'Derramamento de carga', 'Engavetamento', 'Danos eventuais'.

Há apenas algumas distinções por conta da sensibilidade a maiúsculas e minúsculas. E um sinônimo entre 'Colisão com objeto estático' e 'Colisão com objeto fixo'
___
**`fase_dia`**:
'Pleno dia', 'Plena noite', 'Amanhecer', 'Anoitecer', '(null)',
       nan, 'Plena Noite'.
       
Além da string '(null)' que deve ser tratada, há a divergência na distinção por conta da sensibilidade a maiúsculas e minúsculas.

___
**`condicao_metereologica`**
'Ceu Claro', 'Chuva', 'Nublado', 'Sol', 'Nevoeiro/neblina',
       'Ignorada', 'Vento', 'Granizo', '(null)', 'Neve', nan,
       'Garoa/Chuvisco', 'Céu Claro', 'Nevoeiro/Neblina', 'Ignorado'.

Há o problema na string '(null)'.

___
**`tracado_via`**
'Reta', 'Curva', 'Cruzamento', '(null)', 'Não Informado',
       'Interseção de vias', 'Rotatória', 'Desvio Temporário', 'Viaduto',
       'Ponte', 'Retorno Regulamentado', 'Túnel'
       
Há o problema na string '(null)'.


# **2.0 Limpeza dos dados**

---
**- `data_inversa`**

Esse campo possui 2 tipos de datas: ANO - mes - dia; e dia/mes/ANO. Vamos convertela para inversa, mas em `datetime`, deixando-a como %Y-%m-%d;

In [56]:
df['data_inversa']

2012-04-08    114
21/04/2011    113
24/04/2011    112
2012-07-06    106
2012-12-21     91
             ... 
2020-05-24      3
2019-03-04      2
2020-09-24      2
2020-03-25      2
2019-11-13      1
Name: data_inversa, Length: 5022, dtype: int64

In [57]:
### Correção da vírgula para ponto
df['data_inversa'] = pd.to_datetime(df['data_inversa'])

In [58]:
df['data_inversa']

2012-04-08    114
2011-04-21    113
2011-04-24    112
2012-07-06    106
2012-12-21     91
             ... 
2019-01-30      3
2020-03-25      2
2019-03-04      2
2020-09-24      2
2019-11-13      1
Name: data_inversa, Length: 5022, dtype: int64

---
**- `latitude` e `longitude`**

Esses campos estão como string, e devem ser normalizados para float, a fim de utilizá-los em eventuais plots geográficos.

Como esses dois dados são vigente a partir do ano de 2017, vários valores faltantes aparecerão, devido a concatenação das bases.

In [25]:
### Correção da vírgula para ponto
df['latitude']  = df['latitude'].str.replace(',','.')
df['longitude']  = df['longitude'].str.replace(',','.')

#setei os NaN como 0, se der problema nos plot daí tiramos os fillna
#df['latitude'].fillna(0, inplace=True)
#df['longitude'].fillna(0, inplace=True)

In [26]:
df.filter(['latitude', 'longitude']).dtypes

latitude     object
longitude    object
dtype: object

In [27]:
df['latitude'] = df['latitude'].astype(float)
df['longitude'] = df['longitude'].astype(float)

In [28]:
df.filter(['latitude', 'longitude']).dtypes

latitude     float64
longitude    float64
dtype: object

----
**- `dia_semana`**

Para padronizar, vamos deixar tudo em title (primeira letra maiuscula). E depois deixar os dias sem hífem mesmo

In [29]:
df['dia_semana'].value_counts()

Sexta            21129
Sábado           18021
Quinta           17879
Segunda          17424
Domingo          17050
Quarta           16804
Terça            16344
sexta-feira       2785
domingo           2776
sábado            2754
quinta-feira      2522
segunda-feira     2441
terça-feira       2285
quarta-feira      2282
Name: dia_semana, dtype: int64

In [30]:
df['dia_semana'] = df['dia_semana'].apply(lambda x: str(x).title().split('-')[0])

In [31]:
df['dia_semana'].unique()

array(['Segunda', 'Terça', 'Quarta', 'Quinta', 'Sexta', 'Sábado',
       'Domingo'], dtype=object)

----
**- `uf`**

Não precisa limpar nem normalizar pois já está marcada apenas para o RS

In [32]:
df["uf"].value_counts()

RS    142496
Name: uf, dtype: int64

---
**- `br`**

Não precisa limpar nem normalizar(dá pra retirar daqui), está em str

In [33]:
df['br'].unique()

array(['158', '386', '473', '290', '101', '116', '392', '287', '472',
       '293', '153', '471', '285', '468', '480', '377', '604', 116, 290,
       392, 285, 287, 386, 472, 377, 158, 293, 101, 471, 153, 468, 480,
       473, 470, '470', '0', '178', '453', '183', 453, 448, nan],
      dtype=object)

---
**- `causa_acidente`**

In [34]:
#retirar espaços em branco no inicio e final das palavras
seri = pd.Series(df['causa_acidente'])
seri = seri.str.strip()
df['causa_acidente'] = seri

df['causa_acidente'] = df['causa_acidente'].str.title()
df['causa_acidente'] = df['causa_acidente'].str.replace('Falta De Atenção À Condução', 'Falta De Atenção')
df['causa_acidente'] = df['causa_acidente'].str.replace('Defeito Mecânico Em Veículo', 'Defeito Mecânico No Veículo')

df['causa_acidente'].value_counts()

Falta De Atenção                                                                  48602
Outras                                                                            22858
Não Guardar Distância De Segurança                                                17152
Velocidade Incompatível                                                           12994
Desobediência À Sinalização                                                        9214
Ingestão De Álcool                                                                 6654
Defeito Mecânico No Veículo                                                        6163
Animais Na Pista                                                                   4635
Dormindo                                                                           4354
Ultrapassagem Indevida                                                             2942
Desobediência Às Normas De Trânsito Pelo Condutor                                  2382
Defeito Na Via                  

---
**- `tipo_acidente`**

In [35]:
df['tipo_acidente'] = df['tipo_acidente'].str.title()
df['tipo_acidente'] = df['tipo_acidente'].str.replace('Colisão Com Objeto Estático', 'Colisão Com Objeto Fixo')
df['tipo_acidente'].value_counts()

Colisão Traseira                              38088
Saída De Pista                                26253
Colisão Transversal                           20921
Colisão Lateral                               19948
Colisão Com Objeto Fixo                        6549
Tombamento                                     5466
Colisão Frontal                                4846
Saída De Leito Carroçável                      4083
Atropelamento De Animal                        3864
Atropelamento De Pessoa                        2650
Queda De Motocicleta / Bicicleta / Veículo     2497
Capotamento                                    2277
Colisão Com Bicicleta                           955
Colisão Com Objeto Móvel                        889
Danos Eventuais                                 715
Incêndio                                        690
Atropelamento De Pedestre                       685
Derramamento De Carga                           552
Queda De Ocupante De Veículo                    248
Engavetament

---
**- `fase_dia`**

In [36]:
df["fase_dia"] = df["fase_dia"].replace('(null)', np.nan)
df["fase_dia"].fillna("Não Informado", inplace=True)

df['fase_dia'] = df['fase_dia'].str.title()
df['fase_dia'].value_counts()

Pleno Dia        84628
Plena Noite      42203
Anoitecer         8818
Amanhecer         6842
Não Informado        5
Name: fase_dia, dtype: int64

___
**- `condicao_metereologica`**

In [37]:
df["condicao_metereologica"].unique()

array(['Ceu Claro', 'Sol', 'Vento', 'Chuva', 'Nublado',
       'Nevoeiro/neblina', 'Ignorada', '(null)', 'Neve', 'Granizo',
       'Céu Claro', 'Garoa/Chuvisco', 'Nevoeiro/Neblina', 'Ignorado'],
      dtype=object)

In [38]:
from unidecode import unidecode


In [39]:
df["condicao_metereologica"] = df["condicao_metereologica"].replace('(null)', np.nan)
df["condicao_metereologica"].fillna("Não Informado", inplace=True)

df['condicao_metereologica'] = df['condicao_metereologica']\
.str.lower().replace('ignorada', 'nao informado').replace('ignorado', 'nao informado').map(unidecode)

df['condicao_metereologica'].value_counts()

ceu claro           65519
nublado             28755
chuva               23146
sol                 20733
nevoeiro/neblina     2574
nao informado         852
garoa/chuvisco        661
vento                 215
granizo                28
neve                   13
Name: condicao_metereologica, dtype: int64

- **Salvar DataFrame normalizado**

In [40]:
df.to_csv('archive/csv_ocorrencia_normalizado.csv', encoding='latin-1')